### PR TITLE
graph-tool: depend on python-packaging

### PR DESCRIPTION
**Pull request by singingwolfboy**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
                Traceback (most recent call last):
                  File "/private/tmp/pip-build-env-znlqx4af/overlay/lib/python3.11/site-packages/skbuild/setuptools_wrap.py", line 645, in setup
                    cmkr = cmaker.CMaker(cmake_executable)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                  File "/private/tmp/pip-build-env-znlqx4af/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 148, in __init__
                    self.cmake_version = get_cmake_version(self.cmake_executable)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
                  File "/private/tmp/pip-build-env-znlqx4af/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 105, in get_cmake_version
                    raise SKBuildError(msg) from err
  
                Problem with the CMake installation, aborting build. CMake executable is cmake
                [end of output]
```

Maybe missing a direct cmake dependency?

**Comment by SMillerDev** [comment]
Not sure what this could be, should we merge this into the latest boost bump and see if it helps?
```console
==> make install
    CXX      src/graph/draw/libgt_pycairo_aux_la-pycairo_aux.lo
    CXX      src/graph/centrality/graph_betweenness.lo
    CXX      src/graph/centrality/graph_centrality_bind.lo
    CXX      src/graph/centrality/graph_closeness.lo
    CXX      src/graph/centrality/graph_eigentrust.lo
    CXX      src/graph/centrality/graph_eigenvector.lo
  In file included from src/graph/centrality/graph_closeness.cc:23:
  In file included from src/graph/centrality/graph_closeness.hh:24:
  In file included from ./src/graph/histogram.hh:26:
  In file included from /opt/homebrew/include/boost/multi_array.hpp:34:
  In file included from /opt/homebrew/include/boost/multi_array/multi_array_ref.hpp:32:
  /opt/homebrew/include/boost/functional.hpp:45:24: error: no member named 'unary_function' in namespace 'std'; did you mean '__unary_function'?
              using std::unary_function;
                    ~~~~~^
  /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/c++/v1/__functional/unary_function.h:46:1: note: '__unary_function' declared here
  using __unary_function = __unary_function_keep_layout_base<_Arg, _Result>;
  ^
  In file included from src/graph/centrality/graph_closeness.cc:23:
  In file included from src/graph/centrality/graph_closeness.hh:24:
  In file included from ./src/graph/histogram.hh:26:
  In file included from /opt/homebrew/include/boost/multi_array.hpp:34:
  In file included from /opt/homebrew/include/boost/multi_array/multi_array_ref.hpp:32:
  /opt/homebrew/include/boost/functional.hpp:46:24: error: no member named 'binary_function' in namespace 'std'; did you mean '__binary_function'?
              using std::binary_function;
                    ~~~~~^
  /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/c++/v1/__functional/binary_function.h:49:1: note: '__binary_function' declared here
  using __binary_function = __binary_function_keep_layout_base<_Arg1, _Arg2, _Result>;
  ^
  In file included from src/graph/centrality/graph_closeness.cc:23:
  In file included from src/graph/centrality/graph_closeness.hh:24:
  In file included from ./src/graph/histogram.hh:26:
  In file included from /opt/homebrew/include/boost/multi_array.hpp:34:
  In file included from /opt/homebrew/include/boost/multi_array/multi_array_ref.hpp:32:
  /opt/homebrew/include/boost/functional.hpp:180:45: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
          : public boost::functional::detail::unary_function<typename unary_traits<Predicate>::argument_type,bool>
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:214:45: error: no template named 'binary_function' in namespace 'boost::functional::detail'; did you mean '__binary_function'?
          : public boost::functional::detail::binary_function<
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:46:24: note: '__binary_function' declared here
              using std::binary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:252:45: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
          : public boost::functional::detail::unary_function<
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:299:45: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
          : public boost::functional::detail::unary_function<
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:345:57: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
      class mem_fun_t : public boost::functional::detail::unary_function<T*, S>
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:361:58: error: no template named 'binary_function' in namespace 'boost::functional::detail'; did you mean '__binary_function'?
      class mem_fun1_t : public boost::functional::detail::binary_function<T*, A, S>
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:46:24: note: '__binary_function' declared here
              using std::binary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:377:63: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
      class const_mem_fun_t : public boost::functional::detail::unary_function<const T*, S>
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:393:64: error: no template named 'binary_function' in namespace 'boost::functional::detail'; did you mean '__binary_function'?
      class const_mem_fun1_t : public boost::functional::detail::binary_function<const T*, A, S>
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:46:24: note: '__binary_function' declared here
              using std::binary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:438:61: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
      class mem_fun_ref_t : public boost::functional::detail::unary_function<T&, S>
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:454:62: error: no template named 'binary_function' in namespace 'boost::functional::detail'; did you mean '__binary_function'?
      class mem_fun1_ref_t : public boost::functional::detail::binary_function<T&, A, S>
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:46:24: note: '__binary_function' declared here
              using std::binary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:470:67: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
      class const_mem_fun_ref_t : public boost::functional::detail::unary_function<const T&, S>
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:487:68: error: no template named 'binary_function' in namespace 'boost::functional::detail'; did you mean '__binary_function'?
      class const_mem_fun1_ref_t : public boost::functional::detail::binary_function<const T&, A, S>
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:46:24: note: '__binary_function' declared here
              using std::binary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:533:73: error: no template named 'unary_function' in namespace 'boost::functional::detail'; did you mean '__unary_function'?
      class pointer_to_unary_function : public boost::functional::detail::unary_function<Arg,Result>
                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:45:24: note: '__unary_function' declared here
              using std::unary_function;
                         ^
  /opt/homebrew/include/boost/functional.hpp:557:74: error: no template named 'binary_function' in namespace 'boost::functional::detail'; did you mean '__binary_function'?
      class pointer_to_binary_function : public boost::functional::detail::binary_function<Arg1,Arg2,Result>
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  /opt/homebrew/include/boost/functional.hpp:46:24: note: '__binary_function' declared here
              using std::binary_function;
                         ^
  16 errors generated.
  make: *** [src/graph/centrality/graph_closeness.lo] Error 1
  make: *** Waiting for unfinished jobs....
```

File: Formula/b/boost.rb [hunk 1]
```diff
@@ -34,6 +34,13 @@ class Boost < Formula
   uses_from_macos "bzip2"
   uses_from_macos "zlib"
 
+  # Upstream commit for clang 15 compatibility, remove in next version
+  # https://github.com/boostorg/functional/commit/6a573e4
+  patch do
+    url "https://github.com/boostorg/functional/commit/6a573e4.patch?full_index=1"
+    sha256 "c1af12ff3e91cde025874d7dcc6aa24f2beee952d0f3ae70bc3530bf3009cb11"
+  end
```
Comment (chenrui333) [inline Formula/b/boost.rb:42]: looks like the patch is not needed?

```
==> Applying 6a573e4.patch patch -g 0 -f -p1 -i /tmp/boost--patch-20231021-3922-1flzlby/6a573e4.patch can't find file to patch at input line 14 Perhaps you used the wrong -p or --strip option? The text leading up to this was: -------------------------- |From 6a573e4b8333ee63ee62ce95558c3667348db233 Mon Sep 17 00:00:00 2001 |From: Glen Fernandes <glen.fernandes@gmail.com> |Date: Mon, 17 Apr 2023 06:59:02 -0400 |Subject: [PATCH] Define unary_function and binary_function unconditionally | |--- | include/boost/functional.hpp | 7 ------- | 1 file changed, 7 deletions(-) | |diff --git a/include/boost/functional.hpp b/include/boost/functional.hpp |index 644307847087b0bc587a1ca9490b4b85e811d072..6c63146f672b369dad31f5f0fa99a7b20dd2dd86 100644 |--- a/include/boost/functional.hpp |+++ b/include/boost/functional.hpp -------------------------- No file to patch. Skipping patch. 2 out of 2 hunks ignored
```

**Comment by SMillerDev** [comment]
Probably needs https://github.com/Homebrew/homebrew-core/pull/151857 before a rebuild

**Comment by fxcoudert** [comment]
There is now a conflict due to https://github.com/Homebrew/homebrew-core/pull/152092

**Comment by SMillerDev** [comment]
Might be best to make matplotlib a separate package to reign in these dependencies. Why this package need `cmake` and ninja....
```console
          Building wheels for collected packages: ninja, patchelf
            Building wheel for ninja (pyproject.toml): started
            Building wheel for ninja (pyproject.toml): finished with status 'done'
            Created wheel for ninja: filename=ninja-1.11.1.1-cp311-cp311-linux_x86_64.whl size=142699 sha256=96d35598a94fcfc7c653a048bb17cba86d88586bc43513050b22d0ef0ef3cbee
            Stored in directory: /tmp/graph-tool-20231023-23303-196m9gu/graph-tool-2.58/.brew_home/.cache/pip/wheels/2c/51/74/96551d6311f19f633cd825382459851b8e04a9555253538f2b
            Building wheel for patchelf (pyproject.toml): started
            Building wheel for patchelf (pyproject.toml): finished with status 'error'
            error: subprocess-exited-with-error
  
            × Building wheel for patchelf (pyproject.toml) did not run successfully.
```

**Pull request closed without merging**